File: yokochi/src/0411_ALDS1_1_A/Main.java
import java.util.Scanner;

public class Main {
    static Scanner sc = new Scanner(System.in);
    static int n;
    static int a[];

    public static void main(String[] args) {
        read();
        output();
        solve();
    }

    public static void read() {
        n = sc.nextInt();
        a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
    }

    public static void solve() {
        for (int i = 1; i < n; i++) {
            int v = i;
            while (v > 0) {
                if (a[v - 1] > a[v]) {
                    int tmp = a[v];
                    a[v] = a[v - 1];
                    a[v - 1] = tmp;
                }
                v--;
            }
            output();
        }
    }

    public static void output() {
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i]);
            if (i + 1 < a.length) {
                System.out.print(" ");
            }
        }
        System.out.println();
    }
}

File: yokochi/src/0411_ALDS1_1_A/main.cpp
//
//  main.cpp
//  hello
//
//  Created by 横尾　千明 on 2016/04/09.
//
//

#include <iostream>
using namespace std;
int n;
int a[1000];
void output(){
    for(int i=0;i<n;i++){
        cout<<a[i];
        if(i+1<n){
            cout<<" ";
        }
    }
    cout<<endl;
}

int main() {
    //read
    cin>>n;
    for(int i=0;i<n;i++){
        cin>>a[i];
    }
    output();
    //solve
    for(int i=1;i<n;i++){
        int v = i;
        while(v>0&&a[v-1]>a[v]){
                int tmp = a[v];
                a[v]=a[v-1];
                a[v-1]=tmp;
                v--;
        }
        output();
    }
}

